sympy/core/mul.py

```diff
@@ -1196,8 +1196,17 @@ def _eval_is_algebraic_expr(self, syms):
 
     _eval_is_commutative = lambda self: _fuzzy_group(
         a.is_commutative for a in self.args)
-    _eval_is_complex = lambda self: _fuzzy_group(
-        (a.is_complex for a in self.args), quick_exit=True)
+
+    def _eval_is_complex(self):
+        comp = _fuzzy_group((a.is_complex for a in self.args))
+        if comp is False:
+            if any(a.is_infinite for a in self.args):
+                if any(a.is_zero for a in self.args):
+                    return S.NaN.is_infinite
```

Comment:
Should this be `S.NaN.is_finite`?

Reply:
Both are None. I think this should just be None. I've changed this:
```diff
diff --git a/sympy/core/mul.py b/sympy/core/mul.py
index d0f1d807b3..165d97a7c2 100644
--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1201,9 +1201,7 @@ def _eval_is_complex(self):
         comp = _fuzzy_group((a.is_complex for a in self.args))
         if comp is False:
             if any(a.is_infinite for a in self.args):
-                if any(a.is_zero for a in self.args):
-                    return S.NaN.is_infinite
-                if any(a.is_zero is None for a in self.args):
+                if any(a.is_zero is not False for a in self.args):
                     return None
                 return False
         return comp
```